File: initial_value_problem/initial_value.cpp
/*** starting_value.cpp start ***/

#include <fstream>
#include <string>
#include <cmath>

int main(int argc, char* argv[]){
	// starting values, default is (x,y) = (0,1)
	double x0 = 2;
	double y0 = 1;

	// parameters for numerical computation
	// steps is the amount of iterations
	// h is the step width for each iteration
	unsigned int steps = 1000;
	double h = 0.001;

	// running parameters for x and y
	double x = x0;
	double y = y0;
	double yp = 0;

	// filename for output file
	std::string filename = "starting_value.csv";

	// opening output file
	std::ofstream output;
	output.open(filename);

	// printing information about the csv in first line
	output << "x,y,yp\n";

	// iterating with eulers method
	for(unsigned int i = 0; i < steps; i++){
		x = x0 + i * h;
		
		// here yp gets defined
		yp = exp(-1 * x) / sin(x) - y * cos(x) / sin(x);
		//
		y = y + yp * h;

		// storing the data in the file
		output << x << "," << y << "," << yp << "\n";
	}

	output.close();

	return 0;
}

/*** starting_value.cpp end ***/

File: initial_value_problem/initial_value.py
# plot.py start

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

df = pd.read_csv('starting_value.csv')

x = df['x']
y = df['y']
yp = df['yp']

def f(x):
    xi =  ((1 + np.exp(-1 * np.pi / 4)) / np.sqrt(2))
    l = (np.sin(x) + np.cos(x)) * np.exp(-x) / 2
    kl = xi - l
    return kl / np.sin(x)

ya = x.copy()

for i in range(0,len(ya)):
    ya[i] = f(ya[i])

plt.plot(x,y)
plt.plot(x,ya)
plt.show()

# plot.py end
